knx/dpt/formats.go

```diff
@@ -208,6 +208,63 @@ func unpackV8(data []byte, i *int8) error {
 	return nil
 }
 
+func packV16(i int16) []byte {
+	b := make([]byte, 3)
+
+	b[0] = 0
+	b[1] = byte((i >> 8) & 0xff)
+	b[2] = byte(i & 0xff)
+
+	return b
+}
+
+func unpackV16(data []byte, i *int16) error {
+	if len(data) != 3 {
+		return ErrInvalidLength
+	}
+
+	*i = int16(data[1])<<8 | int16(data[2])
+
+	return nil
+}
+
+func packV32AsV16(i int32, resolution int32) []byte {
+	return packV16(int16(i / resolution))
```

Comment:
There are no checks in place here to ensure the value provided (`i`) is within valid bounds. I.e. within `math.MinInt16 * resolution ... math.MaxInt16 * resolution`. 

However, the pack functions does not offer an `error` as return value, and changing this would break the interface and cause ripple effects.